packages/zero-client/src/client/error.ts

```diff
@@ -77,3 +79,94 @@ export function isBackoffError(ex: unknown): BackoffBody | undefined {
 export function isClientError(ex: unknown): ex is ClientError {
   return ex instanceof ClientError;
 }
+
+/**
+ * Returns the status to transition to, or null if the error
+ * indicates that the connection should continue in the current state.
+ */
+export function getErrorConnectionTransition(ex: unknown) {
+  if (isClientError(ex)) {
+    switch (ex.kind) {
+      // Connecting errors that should continue in the current state
+      case ClientErrorKind.AbruptClose:
+      case ClientErrorKind.CleanClose:
+      case ClientErrorKind.ConnectTimeout:
+      case ClientErrorKind.PingTimeout:
+      case ClientErrorKind.PullTimeout:
+      case ClientErrorKind.Hidden:
+        return {status: null, reason: ex} as const;
+
+      // Fatal errors that should transition to error state
+      case ClientErrorKind.NoSocketOrigin:
```

Comment:
I don't think this should result in an error state, since it's a pattern to do local-only queries with server:null. How does this fit into the existing connection API?